README.md

```diff
@@ -1 +1,208 @@
+
+**Example using `content_types` argument:**
+
+```python
+>>> from xtractmime import extract_mime
+>>> content_types = (b'text/xml',)
+>>> body = b''
+>>> extract_mime(body, content_types=content_types)
+b'text/xml'
+```
+
+Optional `http_origin` argument is a flag to determine if the resource is retrieved via HTTP or not.
+`http_origin` is *`True`* (by default) for HTTP responses else *`False`*.
+
+**Example using `http_origin` argument:**
+
+```python
+>>> from xtractmime import extract_mime
+>>> body = b'\x00\x01\xff'
+>>> content_types = (b'text/plain',)
+>>> extract_mime(body, content_types=content_types, http_origin=True)
+b'application/octet-stream'
+>>> extract_mime(body, content_types=content_types, http_origin=False)
+b'text/plain'
+```
+
+Optional `no_sniff` argument is a flag which is *`True`* if the user agent does not wish to
+perform sniffing on the resource and *`False`* (by default) otherwise. The flag is suitable
+for using as a `X-Content-Type-Options` header.
+
+**Example using `no_sniff` argument:**
+
+```python
+>>> from xtractmime import extract_mime
+>>> body = b'''<?xml version="1.0" encoding="utf-8"?>
+                        <rdf:RDF
+                          xmlns:content="http://purl.org/rss/1.0/modules/content/"
+                          xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+                          xmlns="http://purl.org/rss/1.0/"
+                        >
+                        </rdf:RDF>'''
+>>> content_types = (b'text/html',)
+>>> extract_mime(body, content_types=content_types, no_sniff=False)
+b'application/rss+xml'
+>>> extract_mime(body, content_types=content_types, no_sniff=True)
+b'text/html'
+```
+
+Optional `extra_types` argument is a tuple which users may implicitly pass to `xtractmime` for considering additional MIME types
+in determining the MIME type of the provided content. Each entry in the tuple should follow the format
+**(Byte Pattern, Pattern Mask, Leading Bytes, MIME type)**
+
+* **Byte Pattern** is a byte sequence used as a template to be matched against in the pattern matching algorithm. 
+* **Pattern Mask** is a byte sequence used to determine the significance of bytes being compared against a byte pattern in the pattern matching algorithm.
+* **Leading Bytes** is set of bytes to be ignored while matching the leading bytes in the content.
+
+---
+**NOTE**
+
+In a pattern mask, `0xFF` indicates the byte is strictly significant, `OxDF` indicates that the byte is significant in an ASCII case-insensitive way, and 0x00 indicates that the byte is not significant. 
+
+---
+
+**Example using `extra_types` argument:**
+```python
+>>> from xtractmime import extract_mime
+>>> extra_types = ((b'test', b'\xff\xff\xff\xff', None, b'text/test'),)
+>>> extract_mime(b'test', extra_types=extra_types)
+b'text/test'
+>>> extract_mime(b"test")
+b'text/plain'
+```
+
+Optional `supported_types` argument is a set of MIME types that are supported by the user agent.
```

Comment:
Here we need to clarify that the argument is not _a_ set of supported MIME types, but the set of _all_ supported MIME types, possibly with a link to https://mimesniff.spec.whatwg.org/#supported-by-the-user-agent, which defines what a supported MIME type entails. And similarly, clarify that, if no supported types are specified, all MIME types are assumed to be supported.

Also, I think that we should give users a reason to use this option. We should not get into implementation details, but I think it would be safe to indicate that using this parameter _can_ improve xtractmime performance.